clients/redshift/redshift.go

```diff
@@ -91,7 +92,34 @@ WHERE
 }
 
 func (s *Store) Dedupe(fqTableName string) error {
-	return fmt.Errorf("dedupe is not yet implemented")
+	stagingTableName := fmt.Sprintf("%s_dedupe_staging_%.5d", constants.ArtiePrefix, rand.Intn(100_000))
+
+	query := fmt.Sprintf(`
+BEGIN TRANSACTION;
```

Comment:
Use the built-in transactions for go/sql? https://github.com/artie-labs/transfer/blob/a857a7fd9521bb14933270483279185444f81aa5/clients/shared/merge.go#L143

That way we can commit and rollback if there's an error

Reply:
Oh yeah def good call.